### PR TITLE
1.14.9: Check auxpow PoW before the auxpow itself.

### PR DESCRIPTION
**Pull request by patricklodder**
1.14.9 backport of #3725

### CODE REVIEW

**Review by chromatic — APPROVED** [review]
ACK, all tests pass on x86-64 Linux. Given the urgent tag, we'll hold this PR open for an hour for other comments. After that, any maintainer can merge if I don't do so first.

**Comment by michilumin** [comment]
Running a quick testrun on Darwin.

**Review by michilumin — APPROVED** [review]
logic follows.  ack.

**Pull request merged**
